credentials.go

```diff
@@ -133,6 +133,11 @@ func getCredentialsProvider(ctx context.Context, c *Config) (aws.CredentialsProv
 		return nil, "", fmt.Errorf("loading configuration: %w", err)
 	}
 
+	if c.WebIdentityToken != "" && c.AssumeRole != nil {
```

Comment:
Let me know if you'd like to additionally validate/error when AssumeRole is nil when WebIdentityToken is passed.